shared/rocroller/client/src/gemm.cpp

```diff
@@ -1206,10 +1276,16 @@ int main(int argc, const char* argv[])
         "--workgroup_size_x", solution.workgroupSizeX, "Workgroup size in the x dimension.");
     app.add_option(
         "--workgroup_size_y", solution.workgroupSizeY, "Workgroup size in the y dimension.");
-    app.add_option(
-        "--workgroupMapping",
-        [&solution](auto& args) -> bool { return ParseWorkgroupMapping(args[0], solution); },
-        "Workgroup mapping dimension and size.");
+
+    app.add_option("--workgroupMappingDim",
+                   solution.workgroupMappingDim,
+                   "Workgroup mapping dimension (-1, 0, 1). Default: -1")
+        ->check(CLI::IsMember({-1, 0, 1}));
+    app.add_option("--workgroupMappingValue",
+                   runParams.workgroupMappingValue,
+                   "Workgroup mapping value. Default: -1")
+        ->check(CLI::Range(-1, -1) | CLI::PositiveNumber);
```

Comment:
```suggestion
        ->check(CLI::IsMember(-1) | CLI::PositiveNumber);
```
Also, maybe there is a coherent max for the wgm value that should be constrained here.

Reply:
Changed `Range` to `IsMember`. 

I didn't see there is a max number of workgroups allowed in spec, but I found `clinfo` can show max workgroup size: https://github.com/ROCm/ROCm/issues/1388. However, I suspect this might be arch-dependent?   Please let me know if you have a preferred number, or I can use 256 as the max of number of workgroups for now.

Reply:
That can be in QA or a follow up task.